standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ExceptionHandler.java

```diff
@@ -127,7 +128,7 @@ public ExceptionHandler toMetaExceptionIfInstance(String message, Class<?>... cl
    */
   public static void rethrowException(Exception e) throws TException {
     throw handleException(e)
-        .throwIfInstance(MetaException.class, NoSuchObjectException.class)
+        .throwIfInstance(MetaException.class, NoSuchObjectException.class, InvalidObjectException.class)
```

Comment:
nit: This method is invoked elsewhere besides `get_partitions_by_names`, so the same problem would reoccur if some other methods doesn't specify throwing`InvalidObjectException`. 
The original would throw MetaException instead.

Reply:
May be explicitly call `
throw handleException(e) .throwIfInstance(MetaException.class,NoSuchObjectException.class,InvalidObjectException.class)
` instead of modifying rethrowException ?

Reply:
The reason why I explicitly changed this definition is that the HMS server should always throw the right exception to the client. If the InvalidObjectException is not declared in the HMS APIs then we should do so thereby letting know the client about the underlying cause. If we are disguising the InvalidObjectException in the form of MetaException, then I think we are throwing the incorrect exception.
Do you still think we need to modify the way we throw the exception in get_partition_by_names_req() API instead of changing it globally?

Reply:
The `rethrowException` is called in multiple APIs, we should make sure these APIs able to throw InvalidObjectException, otherwise the client will only see `TApplicationException`, which makes things worse. 
The `MetaException` at least will convey the message of `InvalidObjectException` to the client.
I think we'd better modify the way `get_partition_by_names_req()` handles the exception.